app.rb

```diff
@@ -0,0 +1,87 @@
+module Enumerable
+  def my_each
+    i = 0
+    while i < length
+      yield self[i]
+      i += 1
+    end
+  end
+
+  def my_each_with_index
+    i = 0
+    while i < length
+      yield self[i], i
+      i += 1
+    end
+  end
+
+  def my_select
+    new_numbers = []
+    each do |i|
+      new_numbers.push(i) if yield(i)
+    end
+    new_numbers
+  end
+
+  def my_all?
+    condition = true
+    each do |j|
+      condition = false unless yield(j)
+    end
+    condition
+  end
+
+  def my_any?
+    each do |j|
+      return true if yield(j)
+    end
+    false
+  end
+
+  def my_none?
+    condition = true
+    each do |j|
+      condition = false if yield(j)
+    end
+    condition
+  end
+
+  def my_count(counted)
+    counting = 0
+    each do |j|
+      counting += 1 if j == counted
+    end
+    counting
+  end
```

Comment:
1. Should return the number of items in enum through enumeration. For example:
`[1, 2, 3].my_count` - should return `3`

2. Should return the number of items in enum through enumeration::range. For example:
`(1..5).my_count` - should return `5`

3. Should count the number of elements yielding a true value if a block is given. For example:
`[1, 2, 3, 4, 5].my_count {|num| num > 3}` - should return `2`

4. Should not mutate the original array. For example:
`[1, 2, 3].count{ |num| num + 1 }` - should return `3`